### DISCUSSION TITLE
Open specific YouTube video

### DISCUSSION THREAD

**Discussion by anqdc** [maxileith/homebridge-appletv-enhanced#241] [General]
Hello! 

I've been using Apple TV Enhanced now for a few weeks and it's been great, thanks for making it so awesome!

I've made a few shortcuts to different TV shows on Disney+ etc so as to automate when we watch them, but I'm having trouble getting this to work for YouTube videos... Is it possible? If so, how? I've tried copying all sorts of different links, but none of them seem to do the trick!

Anyway, I'd appreciate any help you can give me!

Cheers,
Albert

**Comment by maxileith** [comment]
See https://github.com/maxileith/homebridge-appletv-enhanced/issues/120#issuecomment-1854674611.

**Reply by anqdc** [reply]
Ah I see... So it's just a question of waiting for apple to sort it yeah? 

Any idea how I can play a custom video without having to do the old "left, left, down down, right, select" technique? 

Thanks for your quick reply!

**Reply by maxileith** [reply]
> Any idea how I can play a custom video without having to do the old "left, left, down down, right, select" technique?

No idea what you are talking about.

**Reply by anqdc** [reply]
Haha sorry, I understand myself... :D 

So I want to play a specific video (one of my own videos) on YouTube. For now the way I do it is by running an Automation in HomeKit, converting it to a shortcut and then doing this : 

<img width="285" alt="Screenshot 2024-02-09 at 18 35 10" src="https://github.com/maxileith/homebridge-appletv-enhanced/assets/77596888/1609cb55-b9fa-423f-b80d-859d6770a91d">

Which obviously isn't ideal, cause each action takes about a second, it's just long... So my question was is there any other way (another app maybe) where I can upload my own videos and then use the Apple TV Enhanced app to start playing directly without having to do all that 😂

Or maybe there's a way to do what I've done but much easier and quicker 🤷🏼‍♂️

**Reply by maxileith** [reply]
You could use vlc to stream the video from a NAS or something like that. VLC does handle events on the Apple TV. So opening it will probably work. You just have to figure out how the URL needs to look like. In the README is an example for a VLC custom input.
